blocks/filter.py

```diff
@@ -96,6 +96,16 @@ class object contained in the list.
     """
     def __init__(self, roles=None, bricks=None, each_role=False, name=None,
                  name_regex=None, applications=None):
+        if bricks is not None and not isinstance(bricks, Brick):
```

Comment:
My understanding of that is that you can pass in a class object such as
simply "Linear", and it will give you back all instances of Linear. Are you
saying that this can be accomplished without isclass? Pretty sure you will
get a TypeError if arg 2 to isinstance is not a [sequence of] type/class
objects.

On Fri, Apr 3, 2015 at 5:45 PM, Francesco <notifications@github.com> wrote:

> In blocks/filter.py
> <https://github.com/bartvm/blocks/pull/578#discussion_r27758057>:
>
> > @@ -96,6 +96,16 @@ class object contained in the list.
> >      """
> >      def __init__(self, roles=None, bricks=None, each_role=False, name=None,
> >                   name_regex=None, applications=None):
> > +        if bricks is not None and not isinstance(bricks, Brick):
>
> I guess that isclass can be avoided here
> https://github.com/bartvm/blocks/blob/67e19f33ccdd2d21482fcfce89542a60c0385bad/blocks/filter.py#L131
>
> Also, I am not sure if something can be changed here:
> https://github.com/bartvm/blocks/blob/67e19f33ccdd2d21482fcfce89542a60c0385bad/blocks/filter.py#L147-L150
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bartvm/blocks/pull/578/files#r27758057>.
>